recipes/cc65/all/conanfile.py

```diff
@@ -0,0 +1,182 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, MSBuild, tools
+from conans.client.tools.win import msvs_toolset
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+
+
+class Cc65Conan(ConanFile):
+    name = "cc65"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://cc65.github.io/"
+    description = "A freeware C compiler for 6502 based systems"
+    license = "zlib"
+    topics = ("conan", "cc65", "compiler", "cmos", "6502", "8bit")
+    exports_sources = "patches/**"
+
+    settings = "os_build", "arch_build", "compiler"
+
+    _autotools = None
+    _source_subfolder = "source_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.arch_build not in ("x86", "x86_64"):
+                raise ConanInvalidConfiguration("Invalid arch_build")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and "CONAN_BASH_PATH" not in os.environ \
+                and tools.os_info.detect_windows_subsystem() != "msys2":
+            # msys2 provides make for MSVC and mingw + install for mingw
+            self.build_requires("msys2/20190524")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    @property
+    def _datadir(self):
+        return os.path.join(self.package_folder, "bin", "share", "cc65")
+
+    @property
+    def _samplesdir(self):
+        return os.path.join(self.package_folder, "samples")
+
+    @contextmanager
+    def _mock_settings(self):
+        """
+        This will temporarily copy the attributes arch_build and os_build of settings to arch and os.
+        Because cc65 does not provide x86_64 support for visual studio, x86 is built instead.
+        """
+        mock_settings = self.settings.copy()
+
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.arch_build == "x86_64":
+                self.output.info("x86_64 detected: building x86 instead")
+                mock_settings.arch_build = "x86"
+
+        mock_settings._data["arch"] = mock_settings._data["arch_build"].copy()
+        mock_settings._data["os"] = mock_settings._data["os_build"].copy()
+
+        original_settings = self.settings
+        self.settings = mock_settings
+        yield
+        self.settings = original_settings
```

Comment:
In Python, use always a try/catch/finally approach, just in case:

```suggestion
        try:
            self.settings = mock_settings
            yield
        finally:
            self.settings = original_settings
```